plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -138,6 +140,68 @@ abstract class MavenPublishBaseExtension(
     }
   }
 
+  /**
+   * Set the Maven coordinates consisting of [groupId], [artifactId] and [version] for this project. In the case of
+   * Kotlin Multiplatform projects the given [artifactId] is used together with the platform targets resulting in
+   * artifactIds like `[artifactId]-jvm`.
+   */
+  @Incubating
+  fun coordinates(groupId: String, artifactId: String, version: String) {
+    groupId(groupId)
+    artifactId(artifactId)
+    version(version)
+  }
+
+  private fun groupId(groupId: String) {
+    this.groupId.set(groupId)
+    this.groupId.finalizeValueOnRead()
+
+    // skip the plugin marker artifact which has its own group id based on the plugin id
+    project.mavenPublicationsWithoutPluginMarker {
+      it.groupId = this.groupId.get()
+    }
+  }
+
+  private fun artifactId(artifactId: String) {
+    // skip the plugin marker artifact which has its own artifact id based on the plugin id
+    project.mavenPublicationsWithoutPluginMarker {
+      // the multiplatform plugin creates its own publications, so it is ok to use hasPlugin in here
+      if (project.plugins.hasPlugin("org.jetbrains.kotlin.multiplatform")) {
+        // needed to avoid the mpp plugin writing over our artifact ids
+        project.afterEvaluate { project ->
+          it.artifactId = artifactId.forMultiplatform(it, project)
+        }
+      } else {
+        it.artifactId = artifactId
+      }
```

Comment:
Restructured this a bit. The afterEvaluate is only done for multiplatform and in this case we only set it there instead of setting it regularly and then in afterEvaluate.